packages/sources/dxfeed/src/endpoint/price.ts

```diff
@@ -56,15 +58,19 @@ export const execute: ExecuteWithConfig<Config> = async (request, _, config) =>
 
   // NOTE: may need to force entries quoteEventSymbols to not use batching
 
-  const quotePath = ['Quote', symbol, 'bidPrice']
-  const tradePath = ['Trade', symbol, 'price']
   response.data.result = Requester.validateResultNumber(
     response.data,
-    events === 'Quote' ? quotePath : tradePath,
+    getResultPath(events, symbol),
   )
   return Requester.success(jobRunID, response, config.verbose, batchablePropertyPath)
 }
 
+const getResultPath = (events: Events, symbol: string, isArray = false): Array<string | number> => {
+  const path: Array<string | number> = [events, symbol, events === 'Quote' ? 'bidPrice' : 'price']
+  if (isArray) path.splice(2, 0, 0)
```

Comment:
Dunno about you guys but I find this hard to read.
I suggested a different style in the above comment. How do you feel about something like:
```
  if (Array.isArray(base)) return [events, base, 0, path]
  return [events, base, path]
```
I find this a lot more intuitive.

Reply:
Agreed :+1:

Reply:
Yeah, that looks better IMO